Libplanet.Tests/Blockchain/BlockChainTest.cs

```diff
@@ -2005,10 +2005,82 @@ private async Task FilterLowerNonceTxAfterStaging()
             Assert.Equal(4, _blockChain.StagePolicy.Iterate(_blockChain, filtered: false).Count());
         }
 
+        [Fact]
+        private void CheckIfTxPolicyExceptionHasInnerException()
+        {
+            IValue value = null;
+            var policy = new NullPolicyButTxPolicyAlwaysThrows<DumbAction>(
+                c =>
+                {
+                    // ReSharper disable AccessToModifiedClosure
+                    // The following method calls should not throw any exceptions:
+                    c?.GetStates(new[] { default(Address) });
+                    value = c?.GetState(default);
+                    // ReSharper restore AccessToModifiedClosure
+                });
+            var store = new MemoryStore();
+            var stateStore = new TrieStateStore(new MemoryKeyValueStore());
+            Block<DumbAction> genesisWithTx = MineGenesis(
+                policy.GetHashAlgorithm,
+                GenesisMiner.PublicKey,
+                new[]
+                {
+                    Transaction<DumbAction>.Create(
+                        0,
+                        new PrivateKey(),
+                        null,
+                        Array.Empty<DumbAction>()
+                    ),
+                }
+            ).Evaluate(GenesisMiner, policy.BlockAction, stateStore);
+
+            try
+            {
+                var chain = new BlockChain<DumbAction>(
+                    policy,
+                    new VolatileStagePolicy<DumbAction>(),
+                    store,
+                    stateStore,
+                    genesisWithTx
+                );
+            }
+            catch (TxPolicyViolationException e)
```

Comment:
This `catch` block is called only when `try` block throws exception, but it's not guaranteed. How about using `Assert.Throws<T>()` to insure exception is thrown.
```csharp
Assert.Throws<TxPolicyViolationException>(() => {
    try
    {
        ....
    }
    catch (TxPolicyViolationException e)
    {
        ....

        throw;
    }
});
```

Reply:
I should get used to lambda function, the `Action` (in C#?). 😅